src/test/kotlin/org/rust/ide/inspections/lints/RsPathStatementsInspectionTest.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import org.rust.ProjectDescriptor
+import org.rust.WithStdlibRustProjectDescriptor
+import org.rust.ide.inspections.RsInspectionsTestBase
+
+class RsPathStatementsInspectionTest : RsInspectionsTestBase(RsPathStatementsInspection::class) {
+    fun `test path statement with no effect`() = checkByText("""
```

Comment:
Note, `checkByText` does not check weak warnings by default:
```Kotlin
protected fun checkByText(
    @Language("Rust") text: String,
    checkWarn: Boolean = true,
    checkInfo: Boolean = false,
    checkWeakWarn: Boolean = false,
    ignoreExtraHighlighting: Boolean = false,
) = annotationFixture.checkByText(text, checkWarn, checkInfo, checkWeakWarn, ignoreExtraHighlighting)
```
If we delete `<weak_warning> ... </weak_warning>` from the code, the test still passes. In fact, it is difficult to detect such problems because the only way is to "break" the test and see if it really fails. IMO this API should be changed as it is not clear whether `checkByText` checks certain warnings or not.

@Undin I would either make `org.rust.ide.annotator.RsAnnotationTestBase#checkByText` private or remove the default values for `checkWarn`/`checkInfo`/`checkWeakWarn` parameters to avoid such problems in the future (and maybe even to find existing broken tests). WDYT?